### PR TITLE
clean_dead_slots_from_accounts_index unrefs correctly

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -7586,7 +7586,7 @@ impl AccountsDb {
         let mut accounts_index_root_stats = AccountsIndexRootsStats::default();
         let mut measure = Measure::start("unref_from_storage");
         if let Some(purged_stored_account_slots) = purged_stored_account_slots {
-            let len = purged_stored_account_slots.len();
+            let len = purged_slot_pubkeys.len();
```
Comment (apfitzge) [inline runtime/src/accounts_db.rs:7589]: I'll have to look tomorrow. But at first glance it seems like par_chunks might be an option here. That way we can't accidentally have the wrong number of batches

Comment (apfitzge) [reply runtime/src/accounts_db.rs:7589]: Ah, nevermind. It's a hashset, so we can't use the chunks interface.

**Review by apfitzge — APPROVED** [review]
Have been running a validator, this seems to have fixed the issue. Great work!

**Comment by jeffwashington** [comment]
introduced here:
#20411

**Comment by jeffwashington** [comment]
> will add a test later

To clarify, this function is getting refactored. It will get tests added.

**Pull request merged**